webapp/channels/src/components/post_reminder_custom_time_picker_modal/post_reminder_custom_time_picker_modal.tsx

```diff
@@ -26,7 +26,7 @@ type Props = PropsFromRedux & {
 
 function PostReminderCustomTimePicker({userId, timezone, onExited, postId, actions}: Props) {
     const currentTime = getCurrentMomentForTimezone(timezone);
-    const initialReminderTime = getRoundedTime(currentTime);
+    const initialReminderTime = getReminderRoundedTime(currentTime);
```

Comment:
This seems to work well once you open the modal to pick the custom time. However, when you change the date, it seems to reset back to 12am. 

https://github.com/mattermost/mattermost/assets/936315/d44e7af9-a7b2-4045-9ce9-b61aa0371c44